blobstore/blobstore.smithy

```diff
@@ -292,12 +304,27 @@ structure ObjectMetadata {
 
     /// size of the object in bytes
     @required
-    size: U64,
+    contentLength: U64,
 
     /// date object was last modified
     lastModified: Timestamp,
+
+    /// A MIME type of the object
+    /// see http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.17
+    /// Provider implementations _may_ return None for this field for metadata
```

Comment:
We're starting to get very close to crossing the line between an abstract interface and underlying implementations. I think we're okay here (for now), but we need to be wary of adding anything to the blob interface that cannot be supported by certain types of stores. For example, the file system provider is going to have to attempt to discern mime type and encoding from file extension, which could easily guess wrong.

Reply:
agree. I looked carefully at the data available and thought about utility, and figured it's easier for a provider to keep a mime type map than an actor. If a provider can't easily determine mime type, this api lets them return None.